src/main/java/net/fabricmc/loader/metadata/V1ModMetadataParser.java

```diff
@@ -220,7 +226,29 @@ static LoaderModMetadata parse(Logger logger, JsonReader reader) throws IOExcept
 
 		ModMetadataParser.logWarningMessages(logger, id, warnings);
 
-		return new V1ModMetadata(id, version, environment, entrypoints, jars, mixins, accessWidener, depends, recommends, suggests, conflicts, breaks, requires, name, description, authors, contributors, contact, license, icon, languageAdapters, customValues);
+		return new V1ModMetadata(id, version, aliases, environment, entrypoints, jars, mixins, accessWidener, depends, recommends, suggests, conflicts, breaks, requires, name, description, authors, contributors, contact, license, icon, languageAdapters, customValues);
+	}
+
+	private static void readAliases(JsonReader reader, List<String> aliases) throws IOException, ParseMetadataException {
+		if (reader.peek() != JsonToken.BEGIN_ARRAY) {
+			throw new ParseMetadataException("Aliases must be in an array");
+		}
+
+		reader.beginArray();
+
+		while (reader.hasNext()) {
+			String alias;
+
+			if (reader.peek() != JsonToken.STRING) {
+				throw new ParseMetadataException("Aliases must be a string", reader);
```

Comment:
Since this is referring to a specific entry.

```suggestion
				throw new ParseMetadataException("Alias must be a string", reader);
```